static/js/components/ShareDataForm.jsx

```diff
@@ -0,0 +1,224 @@
+import * as sourceActions from "../ducks/source";
+
+
+const createPhotRow = (id, mjd, mag, magerr, limiting_mag, instrument, filter) => (
+  {
+    id,
+    mjd: Number(mjd).toFixed(3),
+    mag: mag === null ? null : Number(mag).toFixed(4),
+    magerr: magerr === null ? null : Number(magerr).toFixed(4),
+    limiting_mag: Number(limiting_mag).toFixed(4),
+    instrument,
+    filter
+  }
+);
+
+const createSpecRow = (id, instrument) => (
+  {
+    id,
+    instrument
+  }
+);
+
+const photHeadCells = [
+  { name: "id", label: "ID" },
+  { name: "mjd", label: "MJD" },
+  { name: "mag", label: "Mag" },
+  { name: "magerr", label: "Mag Error" },
+  { name: "limiting_mag", label: "Limiting Mag" },
+  { name: "instrument", label: "Instrument" },
+  { name: "filter", label: "Filter" },
+];
+
+const specHeadCells = [
+  { name: "id", label: "ID" },
+  { name: "instrument", label: "Instrument" },
+];
+
+const useStyles = makeStyles(() => ({
+  groupSelect: {
+    width: "20rem",
+  },
+}));
+
+const ShareDataForm = ({ route }) => {
+  const classes = useStyles();
+  const dispatch = useDispatch();
+  const [selectedPhotRows, setSelectedPhotRows] = useState([]);
+  const [selectedSpecRows, setSelectedSpecRows] = useState([]);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const groups = useSelector((state) => state.groups);
+  const photometry = useSelector((state) => state.photometry);
+  const spectra = useSelector((state) => state.spectra);
+  const allGroups = groups.all;
+
+  const { handleSubmit, errors, reset, control, getValues } = useForm();
+
+  useEffect(() => {
+    dispatch(photometryActions.fetchSourcePhotometry(route.id));
+    dispatch(spectraActions.fetchSourceSpectra(route.id));
+  }, [route.id, dispatch]);
+
+  const validateGroups = () => {
+    const formState = getValues({ nest: true });
+    return formState.groups.length >= 1;
+  };
+
+  const onSubmit = async (groupsFormData) => {
+    const selectedPhotIDs = selectedPhotRows.map((idx) => photometry[route.id][idx].id);
+    const selectedSpecIDs = selectedSpecRows.map((idx) => spectra[route.id][idx].id);
+    setIsSubmitting(true);
+    const data = {
+      groupIDs: groupsFormData.groups.map((g) => g.id),
+      photometryIDs: selectedPhotIDs,
+      spectrumIDs: selectedSpecIDs,
+    };
+    const result = await dispatch(sourceActions.shareData(data));
+    if (result.status === "success") {
+      dispatch(showNotification("Data successfully shared"));
+      reset({ groups: [] });
+      setSelectedPhotRows([]);
+      setSelectedSpecRows([]);
+    }
+    setIsSubmitting(false);
+  };
+
+  if (!photometry[route.id] || !allGroups || !spectra[route.id]) {
+    return <>Loading...</>;
+  }
+
+  const photRows = photometry[route.id].map((phot) => (
+    createPhotRow(phot.id, phot.mjd, phot.mag, phot.magerr, phot.limiting_mag,
+      phot.instrument_name, phot.filter)));
+
+  const specRows = spectra[route.id].map((spec) => (
```

Comment:
Just showing the ID and the instrument is not enough to unambiguously identify the spectrum to the user. Perhaps also add the date taken? And an MUI expandable row to show the spectrum plot? Need a way to validate exactly which spectrum is being shared.

Reply:
Date sounds good -- is a plot really necessary here?

Reply:
@dannygoldstein done. Modified the plot handler to optionally take spec ID so we can render the individual spec plots:


![share_spec_plots](https://user-images.githubusercontent.com/7230285/89063579-44ccb480-d31d-11ea-995d-7b354c736683.gif)